File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/memento/Originator.java
package jv.com.tencent.cain.design_pattern.memento;

/**
 * @author : jiangyu
 * @date : 2020/4/17
 * @desc : 备忘录设计模式中的Originator
 */
public class Originator {
    private String state;

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public Memento saveStateMemento() {
        return new Memento(state);
    }

    public void setStateMemento(Memento memento) {
        this.state = memento.getState();
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/memento/Originator.kt
package kt.com.tencent.cain.desigin_pattern.memento

/**
 * @author : jiangyu
 * @date   : 2020/4/17
 * @desc   : 备忘录设计模式中的Originator
 */
class Originator(var state: String = "") {

    fun saveStateMemento(): Memento = Memento(state)

    fun setStateMemento(memento: Memento) {
        this.state = memento.state
    }
}